app/api/v1/vaults.py

```
@@ -51,12 +65,50 @@ def provision_vault(
+    check_access: CheckVaultAccess = Depends(get_check_vault_access_uc),
+    current_user: User = Depends(get_current_user),
 ) -> VaultStatusResponse:
-    use_case = GetVaultStatus(repo)
-    result = use_case.execute(vault_id)
+    use_case = GetVaultStatus(repo, check_access)
+    try:
+        result = use_case.execute(
+            GetVaultStatusInput(vault_id=vault_id, user_id=current_user.id)
+        )
+    except UnauthorizedVaultAccessError as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)